src/wordpiece_tokenizer.cpp

```diff
@@ -94,19 +101,40 @@ bool WordpieceTokenizer::evaluate(ov::TensorVector& outputs, const ov::TensorVec
         new_begins[seq] = ragged_offset;
 
         for(size_t ragged_col = ragged_begins[seq]; ragged_col < ragged_ends[seq]; ++ragged_col) {
-
-            auto str = std::string(chars + begins[ragged_col], chars + ends[ragged_col]);
-            std::vector<core::Token> results = m_tokenizer->Tokenize(str);
-
-            for (const core::Token& token : results) {
+            
+            auto text_orig = std::string(chars + begins[ragged_col], chars + ends[ragged_col]);
+            auto text = text_orig;
+
+            auto text_vec = std::vector<unsigned char>(text.begin(), text.end());
+            int idx = 0;
+            auto token_id = m_trie_root->find_longest(text_vec, idx);
+            int32_t beginning_offset = ragged_offset;
+            if (token_id == -1) {
                 OPENVINO_ASSERT(ragged_offset < outputs[2].get_size());
-                new_elems[ragged_offset++] = token.id_;
-            };
+                new_elems[ragged_offset++] = m_unk_token_id;
+            } else {
+                OPENVINO_ASSERT(ragged_offset < outputs[2].get_size());
+                new_elems[ragged_offset++] = token_id;
+                
+                for(; idx < text_vec.size();) {
+                    text = m_suffix_indicator + text.substr(idx);
+                    text_vec = std::vector<unsigned char>(text.begin(), text.end());
+                    idx = 0;
+                    token_id = m_trie_subwords->find_longest(text_vec, idx);
```

Comment:
Subwords trie shouldn't contain suffix indicator, this way we can avoid input modification.

Reply:
also thought about that. I'll improve that

Reply:
done